test/e2e/router_compression_test.go

```diff
@@ -0,0 +1,261 @@
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	ingress "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestRouterCompressionParsing(t *testing.T) {
+	t.Helper()
+
+	// Test compression policies for the ingress config
+	mimeTypesNormative := []operatorv1.CompressionMIMEType{"text/html", "application/json", "x-custom/allow-custom"}
+	compressionPolicyNormative := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesNormative}
+
+	mimeTypesEmpty := []operatorv1.CompressionMIMEType{}
+	compressionPolicyEmpty := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesEmpty}
+
+	mimeTypesNeedQuotes := []operatorv1.CompressionMIMEType{`text/html; v="keepquoted"`, `x-custom/allow-custom; specialChar='`}
+	compressionPolicyNeedQuotes := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesNeedQuotes}
+
+	mimeTypesErrors := []operatorv1.CompressionMIMEType{"text/", "x- /value", "//"}
+	compressionPolicyErrors := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesErrors}
+
+	testParsing(t, "http-compression-1", compressionPolicyNormative, "error testing normal compression policy: %v")
+	testParsing(t, "http-compression-2", compressionPolicyEmpty, "error testing empty compression policy: %v")
+	testParsing(t, "http-compression-3", compressionPolicyNeedQuotes, "error testing compression policy that needs quotes: %v")
+
+	pc4, err := getPrivateController(t, "http-compression-4", dnsConfig.Spec.BaseDomain)
+	if err != nil {
+		t.Fatalf("error getting private controller: %v", err)
+	}
+	if err := testCompressionPolicy(t, "http-compression-4", compressionPolicyErrors); err == nil {
+		t.Errorf("compression policy with errors should have failed but didn't")
+	}
+	defer assertIngressControllerDeleted(t, kclient, pc4)
+}
+
+func testParsing(t *testing.T, name string, policy operatorv1.HTTPCompressionPolicy, errorMsg string) {
+	t.Helper()
+
+	pc, err := getPrivateController(t, name, dnsConfig.Spec.BaseDomain)
+	if err != nil {
+		t.Fatalf("error getting private controller: %v", err)
+	}
+	if err := testCompressionPolicy(t, name, policy); err != nil {
+		t.Errorf(fmt.Sprintf(errorMsg, err))
+	}
+	defer assertIngressControllerDeleted(t, kclient, pc)
+}
+
+func getPrivateController(t *testing.T, privateName string, privateDomain string) (*operatorv1.IngressController, error) {
```

Comment:
Consider renaming this function as `createPrivateController` to convey that it doesn't just get an existing ingresscontroller but rather creates a new one.

Reply:
👍🏼